readthedocs/api/v2/utils.py

```diff
@@ -159,17 +165,16 @@ def delete_versions_from_db(project, version_data):
         type=BRANCH,
         identifier__in=versions_branches,
     )
-    to_delete_qs = to_delete_qs.exclude(uploaded=True)
-    to_delete_qs = to_delete_qs.exclude(active=True)
-    to_delete_qs = to_delete_qs.exclude(slug__in=NON_REPOSITORY_VERSIONS)
-
-    if to_delete_qs.count():
-        ret_val = {obj.slug for obj in to_delete_qs}
-        log.info('(Sync Versions) Deleted Versions: project=%s, versions=[%s]',
-                 project.slug, ' '.join(ret_val))
+
+    ret_val = set(to_delete_qs.values_list('slug', flat=True))
+    if ret_val:
```

Comment:
I think it would be good if you write a comment here about why we don't use `.count()` as it was.

Reply:
I don't think that's necessary, I'm planning in revising the whole code base for places where we use exists or count and later we evaluate the values.